pkg/providers/vault/config.go

```diff
@@ -36,8 +37,13 @@ type SecretConfig struct {
 }
 
 // BuildURL constructs a URL which can be used to identify a HTTP based secret provider
-func (c SecretConfig) BuildURL() (path string) {
-	return fmt.Sprintf("%s://%s:%v%s", c.Protocol, c.Host, c.Port, c.Path)
+func (c SecretConfig) BuildURL(path string) (url string) {
+	return fmt.Sprintf("%s://%s:%v%s", c.Protocol, c.Host, c.Port, path)
```

Comment:
Might we need to check that the path has no junk in it, only forward slashes, are blanks encoded etc. .. Is it like a web url?

Reply:
yea, i could do some check around this